google_photos_takeout_helper/__main__.py

```diff
@@ -233,38 +268,30 @@ def find_duplicates(path: Path, filter_fun=lambda file: True):
 
                 files_by_full_hash[full_hash].append(file)
```

Comment:
`files_by_full_hash` is appended only in `find_dupicates`... 

So `populate_album_map` can be only run when `--keep-duplicates` wasn't set, right?

We can also make some logic to externally run `find_duplicates`...

<sub>I know, all of this options and flags complicate everything :confused: I wanted to keep them in place in case some functions break, and to be able to just disable them, but now they just break existing options :laughing: </sub>

Reply:
> By the way:
> 
> > `Diagram.jpeg.json`
> > `Diagram.jpg`
> 
> This is outstanding 0_o 🔥

This solves that if we want to add it with one artifact. If I have two identical files with different extensions in the same folder, then this is an issue.
```
    # Returns json dict
    def find_json_for_file(file: Path):
        potential_json = list(file.parent.rglob(file.stem + ".*json"))
        if len(potential_json) != 0:
            try:
                with open(potential_json[0], 'r') as f:
                    dict = _json.load(f)
                return dict
            except:
                raise FileNotFoundError(f"Couldn't find json for file: {file}")
        else:
            raise FileNotFoundError(f"Couldn't find json for file: {file}")
```

Reply:
I'm gonna make a suggestion since I don't have much time recently (due to work and my newborn) to look through all the if/else scenarios. 
1) we get rid of --keep-duplicates option (always remove duplicates) --dont-copy (always copy) and --dont-fix options (always fix). I kind of made those assumptions when writing this code tbh.

2) I explain the rest of my PR here and hand off the torch to someone else to take over to handle these options.